Python/peephole.c

```diff
@@ -536,3 +527,45 @@ PyCode_Optimize(PyObject *code, PyObject* consts, PyObject *names,
     PyMem_Free(codestr);
     return code;
 }
+
+/* Perform basic peephole optimizations to components of a code object.
+   The consts object should still be in list form to allow new constants
+   to be appended.
+
+   To keep the optimizer simple, it bails when the lineno table has complex
+   encoding for gaps >= 255.
+
+   Optimizations are restricted to simple transformations occurring within a
+   single basic block.  All transformations keep the code size the same or
+   smaller.  For those that reduce size, the gaps are initially filled with
+   NOPs.  Later those NOPs are removed and the jump addresses retargeted in
+   a single pass. */
+
+PyObject *
+PyCode_Optimize(PyObject *code, PyObject* consts, PyObject *names,
+                PyObject *lnotab_obj)
+{
+    PyObject* bytecode = NULL;
+    PyObject* old_bytecode = code;
+    int compare_bytecode;
+    do {
+        bytecode = old_bytecode;
```

Comment:
This assignment is unneeded.

Reply:
Yup, but it needs a change in the initialization of `bytecode` as `optimize_bytecode_once` cannot receive `NULL`.